src/Symfony/Component/Cache/Adapter/RedisAdapter.php

```diff
@@ -189,17 +191,35 @@ protected function doClear($namespace)
         }
         foreach ($hosts as $host) {
             if (!isset($namespace[0])) {
-                $host->flushDb();
-            } else {
+                $cleared = $host->flushDb() && $cleared;
+                continue;
+            }
+
+            $info = $host->info('Server');
+            $info = isset($info['Server']) ? $info['Server'] : $info;
+
+            if (!version_compare($info['redis_version'], '2.8', '>=')) {
```

Comment:
Just for curiosity, can't you do: `version_compare($info['redis_version'], '2.8', '<')` ?